model/src/main/java/org/hibernate/performance/search/model/application/DomainDataUpdater.java

```diff
@@ -139,16 +137,29 @@ public void updateQuestionsAndAnswers(Session session, int invocation, int quest
 		}
 
 		List<QuestionnaireInstance> instances = new EmployeeRepository( session ).findByDefinition( definition );
-		for ( QuestionnaireInstance instance : instances ) {
-			for ( ClosedAnswer answer : instance.getClosedAnswers() ) {
+		for ( QuestionnaireInstance instance : choose( instances, 2 ) ) {
```

Comment:
Wouldn't it be more reasonable to include this filtering in the `findByDefinition` query above? From what I understand you may very well be loading hundreds of Questionnaire instances, only to use 2.

Reply:
I agree, I'm changing it...